src/Widgets/calendar/Util.vala

```diff
@@ -499,6 +499,24 @@ namespace Util {
         }
     }
 
+    public void set_event_calendar_color (E.SourceCalendar cal, Gtk.Widget widget) {
+        var color = cal.dup_color ();
+
+        string style = """
+            @define-color colorAccent %s;
+        """.printf (color);
+
+        try {
+            var style_provider = new Gtk.CssProvider ();
```

Comment:
I think that we could optimize this here by storing providers in a HashMap and then creating a new provider only if the color is new and doesn't exist in the provider list. Otherwise, we just get the existing provider.